spec/database_admin_spec.rb

```diff
@@ -752,48 +954,103 @@ def confirm_and_execute
       end
 
       it "calls #ask_local_file_options when choosen" do
         expect(subject).to receive(:ask_local_file_options).once
         say "1"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::LOCAL_FILE)
+        expect(subject.backup_type).to eq("local")
       end
 
       it "calls #ask_nfs_file_options when choosen" do
         expect(subject).to receive(:ask_nfs_file_options).once
         say "2"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::NFS_FILE)
+        expect(subject.backup_type).to eq("nfs")
       end
 
       it "calls #ask_smb_file_options when choosen" do
         expect(subject).to receive(:ask_smb_file_options).once
         say "3"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::SMB_FILE)
+        expect(subject.backup_type).to eq("smb")
       end
 
       it "calls #ask_s3_file_options when choosen" do
         expect(subject).to receive(:ask_s3_file_options).once
         say "4"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::S3_FILE)
+        expect(subject.backup_type).to eq("s3")
       end
 
       it "calls #ask_ftp_file_options when choosen" do
         expect(subject).to receive(:ask_ftp_file_options).once
         say "5"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::FTP_FILE)
+        expect(subject.backup_type).to eq("ftp")
       end
 
       it "cancels when CANCEL option is choosen" do
         say "6"
         expect { subject.ask_file_location }.to raise_error signal_error
       end
+
+      context "with localized file upload" do
+        it "displays custom ftp option with blank prompts" do
+          expect(I18n).to receive(:t).with("database_admin.menu_order").and_return(%w(local ftp://example.com/inbox/filename.txt))
+          expect(I18n).to receive(:t).with("database_admin.local").and_return("The Local file")
+          expect(I18n).to receive(:t).with("database_admin.prompts", :default=>{}).and_return({})
+          expect(subject).to receive(:ask_local_file_options).once
+          say ""
+          subject.ask_file_location
+          expect_output <<-PROMPT.strip_heredoc.chomp + " "
+            Backup Output File Name
+
+            1) The Local file
+            2) ftp to example.com
+            3) Cancel
+
+            Choose the backup output file name: |1|
+          PROMPT
+        end
+
+        it "displays custom ftp option with enabled array" do
+          expect(I18n).to receive(:t).with("database_admin.menu_order").and_return(%w(local ftp://example.com/inbox/filename.txt))
+          expect(I18n).to receive(:t).with("database_admin.local").and_return("The Local file")
+          expect(I18n).to receive(:t).with("database_admin.prompts", :default => {}).at_least(:once).and_return(:"example.com" => { :enabled_for => %w(restore backup) })
```

Comment:
_**Pedantic**_:  Wouldn't this be a known quantity of `.once` or `.twice`?  Would save some characters and would be more explicit with those since this shouldn't be a variable number of calls as far as I can tell.

Reply:
Also, since this line is pretty long, could the prompt options be saved in a `let(:prompt_options)` to allow this to be more visable?

Reply:
The actual value for `enabled_for` are different for most tests. I'll make a helper for this.

I'll make the `twice` more explicit